src/mod_muc_sql.erl

```diff
@@ -236,6 +238,37 @@ set_nick(LServer, Host, From, Nick) ->
 	end,
     ejabberd_sql:sql_transaction(LServer, F).
 
+-spec decode_affiliation(Arg :: binary()) -> atom().
+decode_affiliation(<<"owner">>) -> owner;
+decode_affiliation(<<"member">>) -> member;
+decode_affiliation(<<"outcast">>) -> outcast;
+decode_affiliation(<<"muted">>) -> muted;
+decode_affiliation(_) -> none.
+
+-spec encode_affiliation(Arg :: atom()) -> binary().
+encode_affiliation(owner) -> <<"owner">>;
+encode_affiliation(member) -> <<"member">>;
+encode_affiliation(outcast) -> <<"outcast">>;
+encode_affiliation(muted) -> <<"muted">>;
+encode_affiliation(_) -> <<"none">>.
+
+disable_affiliation(Host, LUser) ->
+    ejabberd_sql:sql_query(Host,
+        ?SQL("update user_affiliation "
+        "set version = version + 1, enabled = 0, last_updated = now() "
+        "where user_id = %(LUser)d and enabled = 1")),
+    ets_cache:delete(user_affiliation_cache, LUser),
+    ok.
+
+insert_affiliation(Host, LUser, Affiliation) ->
+    AffiliationBinary = encode_affiliation(Affiliation),
+    ejabberd_sql:sql_query(Host,
+        ?SQL("insert into user_affiliation "
+        "(version, enabled, user_id, affiliation, date_created, last_updated) "
+        "values (0, 1, %(LUser)d, %(AffiliationBinary)s, now(), now())")),
+    ets_cache:update(user_affiliation_cache, LUser, {ok, Affiliation}, fun() -> ok end),
```

Comment:
Maybe we shouldn't cache the affiliation of the sql query fails.  also, why is the tuple `{ok, Affiliation}` stored instead of just `Affiliation`?

Reply:
ets_cache:update requires that the update value is in the form {ok, value} : ```The Val should be in the form of {ok, V} or error. If existing value in the cache matches Val then UpdateFun is not called```